include/velocypack/inspection/LoadInspector.h

```diff
@@ -0,0 +1,311 @@
+/// @author Manuel Pöter
+////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include <optional>
+#include <string_view>
+#include <tuple>
+#include <type_traits>
+
+#include "velocypack/Builder.h"
+#include "velocypack/inspection/Inspector.h"
+#include "velocypack/Iterator.h"
+#include "velocypack/Slice.h"
+#include "velocypack/Value.h"
+
+namespace arangodb::velocypack::inspection {
+
+struct ParseOptions {
+  bool ignoreUnknownFields = false;
+};
+
+struct LoadInspector : InspectorBase<LoadInspector> {
+  static constexpr bool isLoading = true;
+
+  explicit LoadInspector(Builder& builder, ParseOptions options = {})
+      : LoadInspector(builder.slice(), options) {}
+  explicit LoadInspector(Slice slice, ParseOptions options)
+      : _slice(slice), _options(options) {}
+
+  template<class T, class = std::enable_if_t<std::is_integral_v<T>>>
+  [[nodiscard]] Result value(T& v) {
+    try {
+      v = _slice.getNumber<T>();
+      return {};
+    } catch (Exception& e) {
+      return {e.what()};
+    }
+  }
+
+  [[nodiscard]] Result value(double& v) {
+    try {
+      v = _slice.getNumber<double>();
+      return {};
+    } catch (Exception& e) {
+      return {e.what()};
+    }
+  }
+
+  [[nodiscard]] Result value(std::string& v) {
+    if (!_slice.isString()) {
+      return {"Expecting type String"};
+    }
+    v = _slice.copyString();
+    return {};
+  }
+
+  [[nodiscard]] Result value(bool& v) {
+    if (!_slice.isBool()) {
+      return {"Expecting type Bool"};
+    }
+    v = _slice.isTrue();
+    return {};
+  }
+
+  [[nodiscard]] Result beginObject() {
+    if (!_slice.isObject()) {
+      return {"Expecting type Object"};
+    }
+    return {};
+  }
+
+  [[nodiscard]] Result endObject() { return {}; }
+
+  [[nodiscard]] Result beginArray() {
+    if (!_slice.isArray()) {
+      return {"Expecting type Array"};
+    }
+    return {};
+  }
+
+  [[nodiscard]] Result endArray() { return {}; }
+
+  template<class T>
+  [[nodiscard]] Result list(T& list) {
+    if (auto res = beginArray(); !res.ok()) {
+      return res;
+    }
+    std::size_t idx = 0;
+    for (auto&& s : VPackArrayIterator(_slice)) {
+      LoadInspector ff(s, _options);
+      typename T::value_type val;
+      if (auto res = process(ff, val); !res.ok()) {
+        return {std::move(res), std::to_string(idx), Result::ArrayTag{}};
+      }
+      list.push_back(std::move(val));
+      ++idx;
+    }
+    return endArray();
+  }
+
+  template<class T>
+  [[nodiscard]] Result map(T& map) {
+    if (auto res = beginObject(); !res.ok()) {
+      return res;
+    }
+    for (auto&& pair : VPackObjectIterator(_slice)) {
+      LoadInspector ff(pair.value, _options);
+      typename T::mapped_type val;
+      if (auto res = process(ff, val); !res.ok()) {
+        return {std::move(res), "'" + pair.key.copyString() + "'",
+                Result::ArrayTag{}};
+      }
+      map.emplace(pair.key.copyString(), std::move(val));
```

Comment:
Is is always guaranteed that T is a type that is "emplaceable"? I mean, can it never go wrong to be called with a type that you can't call emplace to? And can it be overwitten (try_emplace)?